src/devices/vdml_motors.c

```diff
@@ -31,19 +31,13 @@ static V5_DeviceMotorPid get_pos_pid(uint8_t port) {
 	return ((motor_data_s_t*)registry_get_device(port)->pad)->pos_pid;
 }
 
-static void set_pos_pid(uint8_t port, V5_DeviceMotorPid pos) {
-	motor_data_s_t* data = (motor_data_s_t*)registry_get_device(port)->pad;
-	data->pos_pid = pos;
-}
+static void set_pos_pid(uint8_t port, V5_DeviceMotorPid pos) { /*deprecated*/ }
```

Comment:
Ah, the point of deprecation is that none of the functionality changes. Rather users are warned that the functionality will be going away in the future. All changes to this file and the c++ source file should be reverted.

Reply:
To be fair, I think in this specific instance given the dangers of the functionality we should remove it. Sure, it's a breaking change but a very warranted one nonetheless.

Reply:
if you want to remove them now, just remove them now instead of doing this halfway thing.

my understanding was that we were following what all best practices for API maintenance dictate, which is to use deprecation as a warning that the function will not exist in the future. not that it currently does something different than what's documented.

this change is unlikely to affect many (if any) users anyway, so I personally think there's little harm in just marking as deprecated and leaving the functionality as is

Reply:
> if you want to remove them now, just remove them now instead of doing this halfway thing.
> 
> my understanding was that we were following what all best practices for API maintenance dictate, which is to use deprecation as a warning that the function will not exist in the future. not that it currently does something different than what's documented.
> 
> this change is unlikely to affect many (if any) users anyway, so I personally think there's little harm in just marking as deprecated and leaving the functionality as is

I just have a feeling that despite marking them deprecated and leaving the functionality as is there is bound to be some stupid high school student that decides to run the code regardless and ends up breaking their v5 motors. Given that this outcome even sparsely happening from time to time is terrible, I would still give the go ahead to make the API removal. 

We should definently make it clear to our library writers (specifically rust and okapi) that we are doing this and they should remove any uses of these functions before upgrading their kernels (it'd cause a compilation error anyways).

Reply:
Not that I have much say, but I vote to remove the feature completely. However, if that is not an option, I agree with hotel to keep the functionality with deprecation. With the current change, it is still breaking, since it affects users who use the pid set. Then, it will be impossible to set the pid, yet the method still exists, showing a warning instead of a compile error.

It would make sense to deprecate now and remove in pros 4, but I think removing the feature now is a good option. You could then also remove stuff like `motor_pid_s_t` which is bloating pros, but doing nothing.